replications/remotewrite/writer.go

```diff
@@ -85,91 +86,83 @@ func NewWriter(replicationID platform.ID, store HttpConfigStore, metrics *metric
 	}
 }
 
-func (w *writer) Write(data []byte) error {
-	// Clean up the cancellation goroutine on a successful write.
-	writeDone := make(chan struct{})
+func (w *writer) Write(data []byte, attempts int) (backoff time.Duration, shouldRetry bool, err error) {
+	cancelOnce := &sync.Once{}
+	// Cancel any outstanding HTTP requests if the replicationQueue is closed.
+	ctx, cancel := context.WithCancel(context.Background())
+
 	defer func() {
-		select {
-		case writeDone <- struct{}{}: // send signal if the cancellation goroutine is still waiting to receive
-		default:
-		}
+		cancelOnce.Do(cancel)
 	}()
 
-	// Cancel any outstanding HTTP requests if the replicationQueue is closed.
-	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		select {
 		case <-w.done:
-			cancel()
-		case <-writeDone:
+			cancelOnce.Do(cancel)
+		case <-ctx.Done():
+			// context is cancelled already
 		}
 	}()
 
-	attempts := 0
+	// Get the most recent config on every attempt, in case the user has updated the config to correct errors.
+	conf, err := w.configStore.GetFullHTTPConfig(ctx, w.replicationID)
+	if err != nil {
+		return w.backoff(attempts), true, err
+	}
 
-	for {
+	res, postWriteErr := PostWrite(ctx, conf, data, w.clientTimeout)
+	res, msg, ok := normalizeResponse(res, postWriteErr)
+	if !ok {
+		// bail out
+		return w.backoff(attempts), true, postWriteErr
```

Comment:
I think this code is right, but `normalizeResponse` says the bool is for if the error is retryable. I think the comment on `normalizeResponse` is lying here?